### PR TITLE
Xenomorph updates.

### PR DESCRIPTION
**Pull request by Emanthealmighty**
**What does this PR do:**
This is the redemption arc of my previous PR. I have played around with xenos on my own server and I've noticed just how exceedingly good they are at killing people. They can push people over way too easily, and borgs can be stunlocked by just pushing them, which practically never fails.

The following changes have been made:

Edit: reduced the nerfs a little across the board. The comment below has the updated information.

Decreased the chances of xenos shoving both humans and cyborgs.
Reduced the amount of time you are on the ground when shoved by a xeno as a human.
Reduced the amount of time you are stunned when shoved by a xeno as a cyborg. Can no longer easily be stunlocked.
Reduced the chances of weakening (stunning) a human as a xeno by harming them. However, the stun time for it has been slightly increased.
Xenomorph Queens are now slightly sturdier. (If they can't reach people to beat them up, then they should at least be able to withstand a slightly bigger beating)
Descriptions have been added to every xenomorph.
Xenomorphs now retain their identification number when they evolve from a larva.

**Changelog:**
:cl: EmanTheAlmighty
tweak: Reduced the chances of getting shoved and stunned by a xenomorph as a human or cyborg.
tweak: Reduced the amount of time you are stunned by a xenomorph as a human or cyborg.
tweak: Reduced the chances of stunning a human as a xenomorph by harming them. The stun time has been slightly increased.
tweak: Xenomorph Queens are now slightly sturdier.
add: Descriptions have been given to every xenomorph.
add: Xenomorphs now retain their identification number when they evolve from a larva.
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
How much are the reductions? Should probably list specifics. Sounds decent from face value though.

**Comment by florkeger** [comment]
> How much are the reductions? Should probably list specifics. Sounds decent from face value though.

Just look at the code changes? :dagger:

**Comment by Emanthealmighty** [comment]
> How much are the reductions? Should probably list specifics. Sounds decent from face value though.

The chance of shoving someone as a xeno was 80%. Now it's 40%.
If you failed to shove someone, you had a 90% chance of disarming them. Now it's 75%.

The chance of weakening someone as a xeno was 33%. Now it's 13%.

The chance of stunning a robot was 85%. Now it's 55%.

**Comment by Couls** [comment]
Ehhh tweaking the numbers isn’t that great of a solution really and this might just make people hate playing xenos since they’ve been weakened so much they can’t rven handle a single person

**Comment by AffectedArc07** [comment]
>Just look at the code changes? 🗡

You should be documenting stuff in your PR description, without making people read through your code

**Comment by Mitchs98** [comment]
Precisely what Affected said. You should not need to read the code to understand what the basics of a PR. That said. The changes sound good to me. I disagree with Coul assuming xenos can't handle a single person, as even with those numbers they'd still be pretty strong. 

Xenos shouldn't really be guarunteed to 1 vs 1 prepared crew anyway. Terror Spiders certainly aren't, so why should they?

File: code/modules/mob/living/carbon/alien/humanoid/queen.dm [hunk 1]
```diff
@@ -30,7 +31,7 @@
 
 /mob/living/carbon/alien/humanoid/queen/movement_delay()
 	. = ..()
-	. += 3
+	. += 2
```
Comment (Couls) [inline code/modules/mob/living/carbon/alien/humanoid/queen.dm]: um you still increased the speed of the queen

File: code/modules/mob/living/carbon/alien/humanoid/queen.dm [hunk 2]
```diff
@@ -8,6 +8,7 @@
 	heal_rate = 5
 	large = 1
 	ventcrawler = 0
+	desc = "A bulky alien. This one seems to be a jack of all trades, its only drawback being its size."
```
Comment (Couls) [inline code/modules/mob/living/carbon/alien/humanoid/queen.dm]: inaccurate description, queens aren't the jacks of all trades they're the mothers of the nest and players should know they should stay in the nest and generate eggs

File: code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm [hunk 3]
```diff
@@ -4,6 +4,7 @@
 	maxHealth = 100
 	health = 100
 	icon_state = "aliend_s"
+	desc = "A strange alien. This one seems to be less adept at fighting."
```
Comment (Couls) [inline code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm]: These ones are the builders, they make weeds and nests and walls for the other xenos

Comment (Emanthealmighty) [reply code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm]: I did not want to add gameplay details to any of the descriptions. Giant spiders, for example, when examined, only report the color of their eyes. If it's a green one, it doesn't tell you "this spider makes the eggs and has to wrap someone in a web beforehand", it just says "this one has green eyes".

Examining something at a first glance shouldn't inform you of what exactly something is or does.

File: code/modules/mob/living/carbon/alien/humanoid/queen.dm (hunk 1 above)
Comment (Emanthealmighty) [reply code/modules/mob/living/carbon/alien/humanoid/queen.dm]: That's intentional.

File: code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm (hunk 3 above)
Comment (Couls) [reply code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm]: but if you didn't want to add gameplay details then wouldn't "this one seems to be less adept at fighting" be a gameplay detail and not something you see at first glance?

Comment (Emanthealmighty) [reply code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm]: You can see at first glance that it doesn't seem built for fighting.

Maybe you know why, maybe you don't. You probably heard thay there are more aliens aboard, and they are very likely stronger than this one.

**Comment by Ralta** [comment]
It's nice to see that you've spent some time in a private server testing and tweaking, but I honestly feel your reasoning is flawed. Xenos are supposed to be _significantly_ stronger than humans in one on one combat. They can't stand up to much direct damage, so they have speed and stuns on their side. 

Some of these changes are fine, like borgs not being stunned as long by a push - 85% to 55% seems reasonable to me. But then a 13% to weaken on a harm hit is far too low. If you're going to reduce it, drop it to 25% so that even on repeated harm hits, there's only a 1 in 4 chance. Xenos often don't harm repeatedly when trying to boost their numbers because they don't want to kill potential hosts. Harming threats like an ERT is reasonable, but I would expect them to be weakened after a couple of hits.

You also added the Queen speed buff from your previous PR, which was heavily down-voted. You've indicated they're the 'jack of all trades', which is completely inaccurate, as the Queen serves a very specific non-combat function. And if you want to compare it to the source material, Queens aren't that fast to begin with (when compared with any other caste).

Finally, removing sprites to make adding future sprites is...odd. I completely understand what you're getting at, but removing nice sprites in anticipation of _someone_ spriting new Xeno variants doesn't make any sense. If you had a PR up that added new Xeno variants and didn't sprite the walking, that would be a different discussion. As it stands, this change is too early.

**Comment by Emanthealmighty** [comment]
> It's nice to see that you've spent some time in a private server testing and tweaking, but I honestly feel your reasoning is flawed. Xenos are supposed to be _significantly_ stronger than humans in one on one combat. They can't stand up to much direct damage, so they have speed and stuns on their side.
> 
> Some of these changes are fine, like borgs not being stunned as long by a push - 85% to 55% seems reasonable to me. But then a 13% to weaken on a harm hit is far too low. If you're going to reduce it, drop it to 25% so that even on repeated harm hits, there's only a 1 in 4 chance. Xenos often don't harm repeatedly when trying to boost their numbers because they don't want to kill potential hosts. Harming threats like an ERT is reasonable, but I would expect them to be weakened after a couple of hits.
> 
> You also added the Queen speed buff from your previous PR, which was heavily down-voted. You've indicated they're the 'jack of all trades', which is completely inaccurate, as the Queen serves a very specific non-combat function. And if you want to compare it to the source material, Queens aren't that fast to begin with (when compared with any other caste).
> 
> Finally, removing sprites to make adding future sprites is...odd. I completely understand what you're getting at, but removing nice sprites in anticipation of _someone_ spriting new Xeno variants doesn't make any sense. If you had a PR up that added new Xeno variants and didn't sprite the walking, that would be a different discussion. As it stands, this change is too early.

This here is the kind of answer I would like to see on anybody's PR. Someone who understands why the PR maker thought of changing something, and someone who gives a good reasoning NOT to change something.

Right, to start with: I can up the stun and weaken chance on humans by a little bit. They are still very much capable of 1v1ing someone with ease. I'm happy that you agree with me on the borg stun change.
As you've been the second person now to disagree with the queen's description and insignificant speed buff, I can go ahead and change it and remove the latter.
And finally, about the walk sprites, I personally wanted to do it for a long time. They just look weird and out of place. There is no reason for them to have a different walk sprite as nothing else does. The only thing that needs a different walk sprite is the hunter, and even then it's still just the running sprite, but transparent.

File: code/modules/mob/living/carbon/alien/humanoid/queen.dm (hunk 2 above)
Comment (Emanthealmighty) [reply code/modules/mob/living/carbon/alien/humanoid/queen.dm]: It'll be changed.

**Comment by Ralta** [comment]
> This here is the kind of answer I would like to see on anybody's PR. Someone who understands why the PR maker thought of changing something, and someone who gives a good reasoning NOT to change something.
> 
> Right, to start with: I can up the stun and weaken chance on humans by a little bit. Aliens are still pretty capable of 1v1ing someone with ease. I'm happy that you agree with me on the borg stun change.
> As you've been the second person now to disagree with the queen's description and insignificant speed buff, I can go ahead and change it and remove the latter. (I just wanted to have somethig good among the nerfs)
> And finally, about the walk sprites, I personally wanted to do it for a long time. They just look weird and out of place. There is no reason for them to have a different walk sprite as nothing else does. The only thing that needs a different walk sprite is the hunter, and even then it's still just the running sprite, but transparent.

Happy to try and be constructive, where possible :wink: 

Glad you're willing to find a compromise, but Xenos shouldn't be `pretty capable` in a 1v1, they should absolutely dominate in a stand-up close-range fight with an unarmed human. When the human is armed, it's a different ballgame, but the Xeno should still have a good edge most of the time in 1v1 situations.

I don't think you always need to balance nerfs with buffs. Sometimes, things need to be rebalanced and that can be achieved purely with nerfs. In this case, some of your ideas aren't bad, but require some tweaking to be viable.

As for walking sprites, I rather like them but that's a personal opinion. If, as I mentioned before, someone was prepared to sprite and implement different Xeno types on the proviso that walking sprites were removed, that would be pretty understandable. As it stands, this isn't the case. I would hold off on an unnecessary removal unless more people indicate they also don't like the walking animations.

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 4]
```diff
@@ -24,6 +24,7 @@
 	var/list/alien_organs = list()
 	var/death_message = "lets out a waning guttural screech, green blood bubbling from its maw..."
 	var/death_sound = 'sound/voice/hiss6.ogg'
+	var/idtag = null // the tag next to their name to identify them
```
Comment (Couls) [inline code/modules/mob/living/carbon/alien/alien.dm]: where is this actually used? It doesn't look like any of the xenos have any tags yet

File: code/modules/mob/living/silicon/robot/robot_defense.dm [hunk 5]
```diff
@@ -2,10 +2,9 @@
 	if(M.a_intent == INTENT_DISARM)
 		if(!lying)
 			M.do_attack_animation(src, ATTACK_EFFECT_DISARM)
-			if(prob(85))
-				Stun(7)
-				step(src, get_dir(M,src))
-				spawn(5)
+			if(prob(55))
+				Stun(2)
+				spawn(2)
```
Comment (Couls) [inline code/modules/mob/living/silicon/robot/robot_defense.dm]: not sure why you lowered the length spawn as well

Comment (Emanthealmighty) [reply code/modules/mob/living/silicon/robot/robot_defense.dm]: The spawn was a delay to make the robot move another tile after moving the first time from being shoved. They got shoved two tiles back.

As I changed it to be pushed back only one time, there was no reason to keep such a long delay.

Comment (Couls) [reply code/modules/mob/living/silicon/robot/robot_defense.dm]: you don't actually need a spawn at all then, there's no reason for a delay if there's only the first push

**Comment by Kyep** [comment]
Nerfing xenomorphs is not a good idea.
Xenos are intended to be OP. They're a major event antag. Their purpose is to be a threat to the whole station, like blobs or terror spiders. They're also sort of a horror monster. They need to be scary - something that makes the crew freak out when they're sighted. Nerfing them goes directly against these goals.

**Comment by Emanthealmighty** [comment]
There's difference between being scary and being unbalanced.
Blobs and terror spiders do not possess the ability to disable people and borgs for a long time with an almost non existent fail rate just by getting near them. Yet they are still scary and make people freak out over comms. 

Xenos, on the other hand, stun borgs and people with extreme ease, and can even stun unprepared people from afar by throwing a hugger at them, rendering them unable to speak, which easily grants the xenos more allies.

I can agree to reduce the nerfs by a bit but as it stands they are way too strong, the only case in which they don't win is when something screws them over during their early stages, but there is absolutely nothing to stop them afterwards when they get more xenos, always resulting in a boring nuke or quarantine that prevents ERTs.

With my changes, odds are still in favor of xenos. I have tested it. This isn't a PR that cripples xenos, but rather renders them a bit more fair to play against. It isn't all nerfs, either, as far as combat goes.

**Comment by Kyep** [comment]
I agree with you that the xeno's reliance on stun mechanics, in particular chain-stun, is horrible. While it makes them strong, it makes them strong in the most un-fun way possible. This was part of the reason I made terror spiders with a goal of minimizing stun mechanics.

However, I think if you're going to reduce stun mechanics from xenos.... you need to dramatically rethink how they work. One idea for a quick-fix would be to rebalance them by trading easy chain-stun disarms for something else. For example, you could make it so that xenos on disarm intent don't chain stun you as easily, but being facehugged renders you unconscious for 80% of the 3 minutes it takes you to burst. Additionally, xenos would start with more lava, perhaps 5 larva.... or they'd just start as fully-grown xenos skipping the laval stage, or some combination. Essentially, make their ability to kidnap people more dependent on the use of facehuggers, and less able to use chain-stunning disarm attacks. By virtue of the fact they'd be significantly worse in combat, substantially increase the number of them that spawn at event start, or their initial growth rate. You could even recode xeno facehuggers so they will leap more distance at people, or crawl through vents to specifically target people in other rooms. This would help keep xenos strong and scary, whilst at the same time reducing their reliance on stun-based, un-fun, disarm-spam combat.

I fully support altering xenos in a creative way to make them more fun and less about spamming disarms. 
What I don't support is overall nerfing them.
And I do think this seems like an overall nerf to them. You say you've tested it and that it isn't... but right now xenos are so reliant on spamming disarms in combat you have to justify how your buffs outweight that far more if you don't want this to be considered an objection-worthy overall nerf.

**Comment by Emanthealmighty** [comment]
I am definitely up for changing their mechanics rather than modifying some numbers. 

I mainly wanted to see these simple changes tested to see how it would go, because according to my tests there is still a good chance you'll land a disarm on a person or cyborg, but it might take one or two tries on some attempts. I wanted to see if it would be enough, but changing them up sounds like a better idea.

**Comment by Necaladun** [comment]
Some of this I like - the queen health buffs for instance. However i see xenos having a much more alien 1 feel for this server. The crew arent some kind of marines who are dealing with them. One should be able to easily take out groups of people and a nest of them should be game over man game over. 
Xenos are not a regular round type for good reason, and if a round  event they need to be quickly wiped out. If not then the crew should be fucked, and nuking from orbit is the strategy required. 
I'd love them to be tweaked a hell of a lot more in general but that requires a holistic approach rather than a few piecemeal ones.

I do appreciatete the effort to tweak them and would be interested in chatting about what would be better when i have the time, shoot me a message on discord.

**Pull request closed without merging**